packages/dev/core/src/FlowGraph/flowGraphExecutionBlock.ts

```diff
@@ -0,0 +1,43 @@
+import type { FlowGraph } from "./flowGraph";
+import { FlowGraphBlock } from "./flowGraphBlock";
+import { FlowGraphConnectionPointDirection, FlowGraphSignalConnectionPoint } from "./flowGraphConnectionPoint";
+
+/**
+ * @experimental
+ * A block that executes some action. Always has an input signal.
+ * Can have one or more output signals.
+ */
+export abstract class FlowGraphExecutionBlock extends FlowGraphBlock {
+    public readonly onStart: FlowGraphSignalConnectionPoint;
+
+    private readonly _signalInputs: FlowGraphSignalConnectionPoint[] = [];
+    private readonly _signalOutputs: FlowGraphSignalConnectionPoint[] = [];
+
+    constructor(graph: FlowGraph) {
+        super(graph);
+
+        this.onStart = this._registerSignalInput("onStart");
+    }
+    /**
+     * Executes the flow graph execution block.
+     */
+    abstract execute(): void;
+
+    protected _registerSignalInput(name: string): FlowGraphSignalConnectionPoint {
+        const input = new FlowGraphSignalConnectionPoint();

```

Comment:
Same comment than above for the `FlowGraphDataConnectionPoint`: I would add a constructor to `FlowGraphSignalConnectionPoint` to pass the initial data.